tf2_ros/include/tf2_ros/transform_listener.h

```diff
@@ -29,65 +29,92 @@
 
 /** \author Tully Foote */
 
-#ifndef TF2_ROS_TRANSFORMLISTENER_H
-#define TF2_ROS_TRANSFORMLISTENER_H
+#ifndef TF2_ROS__TRANSFORM_LISTENER_H_
+#define TF2_ROS__TRANSFORM_LISTENER_H_
 
+#include <memory>
+#include <string>
 #include <thread>
 #include "tf2_msgs/msg/tf_message.hpp"
 #include "rclcpp/rclcpp.hpp"
 
 #include "tf2_ros/buffer.h"
 #include "tf2_ros/visibility_control.h"
 
-namespace tf2_ros{
+
+namespace tf2_ros
+{
 
 /** \brief This class provides an easy way to request and receive coordinate frame transform information.
  */
 class TransformListener
 {
-
 public:
   /**@brief Constructor for transform listener */
   TF2_ROS_PUBLIC
-  TransformListener(tf2::BufferCore& buffer, bool spin_thread = true);
-  
+  TransformListener(tf2::BufferCore & buffer, bool spin_thread = true);
+
+  TF2_ROS_PUBLIC
+  TransformListener(tf2::BufferCore & buffer, rclcpp::Node::SharedPtr nh, bool spin_thread = true);
+
   TF2_ROS_PUBLIC
-  TransformListener(tf2::BufferCore& buffer, rclcpp::Node::SharedPtr nh, bool spin_thread = true);
+  TransformListener(
+    tf2::BufferCore & buffer,
+    const rclcpp::node_interfaces::NodeBaseInterface::SharedPtr & node_base,
+    const rclcpp::node_interfaces::NodeTopicsInterface::SharedPtr & node_topics,
+    bool spin_thread = true);
 
   TF2_ROS_PUBLIC
   ~TransformListener();
 
 private:
-
   /// Initialize this transform listener, subscribing, advertising services, etc.
   void init();
   void initThread();
 
+  template<
```

Comment:
this signature should not be necessary. Including [this](https://github.com/ros2/rclcpp/blob/master/rclcpp/include/rclcpp/create_subscription.hpp) header file should be enough